src/components/TodoItem.jsx

```diff
@@ -1,13 +1,74 @@
-const TodoItem = ({ itemProp, handleChange, delTodo }) => {
+import { useState, useRef } from 'react';
+import styles from '@/styles/TodoItem.module.css';
+import { FaTrash } from "react-icons/fa";
+import { AiFillEdit } from "react-icons/ai";
+import { useAuthContext } from '@/context/AuthContext';
+
+const TodoItem = ({ itemProp, handleChange, delTodo, setUpdate }) => {
+  const editInputRef = useRef(null);
+  const [editing, setEditing] = useState(false);
+  const { user } = useAuthContext();
+
+  const completedStyle = {
+    fontStyle: 'italic',
+    color: '#595959',
+    opacity: 0.4,
+    textDecoration: 'line-through',
+  };
+
+  const handleEditing = () => {
+    setEditing(true);
+  };
+
+  let viewMode = {};
+  let editMode = {};
+  if (editing) {
+    viewMode.display = 'none';
+  } else {
+    editMode.display = 'none';
+  }
+
+  const handleUpdatedDone = (event) => {
+    if (event.key === 'Enter') {
+      setUpdate(editInputRef.current.value, itemProp.id);
+      // setUpdateInput(updateInput, itemProp.id);
+      setEditing(false);
+    }
+  };
+
   return (
-    <li>
+    <li className={styles.item}>
+      <div className={styles.content} style={viewMode}>
+        <input
+          type="checkbox"
+          checked={itemProp.completed}
+          onChange={() => handleChange(itemProp.id)}
+        />
+        {user && (
+          <button onClick={handleEditing}>
+            <AiFillEdit 
+              style={{ color: "#5e5e5e", fontSize: "16px" }}
+            />
+          </button>
+        )}
+
+        <button onClick={() => delTodo(itemProp.id)}>
+          <FaTrash style={{ color: "#5e5e5e", fontSize: "16px" }} />
+        </button>
+        <span style={itemProp.completed ? completedStyle : null}>
+          {itemProp.title}
+        </span>
+      </div>
       <input
-        type="checkbox"
-        checked={itemProp.completed}
-        onChange={() => handleChange(itemProp.id)}
+        type="text"
+        // value={updateInput}
+        ref={editInputRef}
+        defaultValue={itemProp.title}
+        className={styles.textInput}
+        style={editMode}
+        // onChange={(e) => setUpdateInput(e.target.value)}
```

Comment:
I think you've forgotten to remove these commented lines.

Reply:
**Thank you @ibugithub** 
👍🏼